src/features/pagination/js/pagination.js

```diff
@@ -146,12 +176,14 @@
             var visibleRows = renderableRows.filter(function (row) { return row.visible; });
             grid.options.totalItems = visibleRows.length;
 
-            var firstRow = (currentPage - 1) * pageSize;
+            var firstRow = publicApi.methods.pagination.getFirstRowIndex();
+            var lastRow  = publicApi.methods.pagination.getLastRowIndex();
+
             if (firstRow > visibleRows.length) {
               currentPage = grid.options.paginationCurrentPage = 1;
               firstRow = (currentPage - 1) * pageSize;
             }
-            return visibleRows.slice(firstRow, firstRow + pageSize);
+            return visibleRows.slice(firstRow, lastRow);
```

Comment:
`firstRow` is `(currentPage - 1) * pageSize`
Previously, `lastRow` was `firstRow + pageSize`, which is algebraically
```
lR = fR + pS          | fR = (cP - 1) * pS
   = (cP-1) * pS + pS | distribute pS over (cP-1)
   = cP*pS - pS + pS  | cancel addition
   = cP*pS
```
With the change, `lastRow` is `currentPage * pageSize` (or `totalItems`), which simply eliminates the math in the middle.